packages/wouter/test/link.test-d.tsx

```diff
@@ -1,44 +1,60 @@
-import { describe, it, assertType } from "vitest";
+import { describe, it } from "vitest";
 import { Link } from "wouter";
 import * as React from "react";
 
-describe("Link types", () => {
+describe("<Link /> types", () => {
   it("should have required prop href", () => {
     // @ts-expect-error
-    assertType(<Link>test</Link>);
-    assertType(<Link href="/">test</Link>);
+    <Link>test</Link>;
+    <Link href="/">test</Link>;
   });
 
-  it("should support state prop", () => {
-    assertType(
-      <Link href="/" state={{ a: "foo" }}>
-        test
-      </Link>
-    );
-    assertType(
-      <Link href="/" state={null}>
-        test
-      </Link>
-    );
-    assertType(
-      <Link href="/" state={undefined}>
-        test
-      </Link>
-    );
-    assertType(
-      <Link href="/" state="string">
-        test
-      </Link>
-    );
+  it("should throw error when `to` and `href` props are used in same time", () => {
+    // @ts-expect-error
+    <Link to="/hello" href="/world">
+      Hello
+    </Link>;
+  });
+
+  it("should inherit props from `HTMLAnchorElements`", () => {
+    <Link to="/hello" className="hello">
+      Hello
+    </Link>;
+
+    <Link to="/hello" style={{}}>
+      Hello
+    </Link>;
+
+    <Link to="/hello" target="_blank">
+      Hello
+    </Link>;
+
+    <Link to="/hello" download ping="he-he">
+      Hello
+    </Link>;
+  });
+
+  it("should support other navigation params", () => {
+    <Link href="/" state={{ a: "foo" }}>
+      test
+    </Link>;
+
+    <Link href="/" replace>
+      test
+    </Link>;
+
+    <Link href="/" state={undefined}>
+      test
+    </Link>;
   });
```

Comment:
Imo, we should also test generic Link type with custom navigation hook. Let's say for example `useNetworkLocation` and `navigate({ host: "whatever", retries: 4 })`.